geolocator_web/lib/src/html_permissions_manager.dart

```diff
@@ -23,12 +22,10 @@ class HtmlPermissionsManager implements PermissionsManager {
       return LocationPermission.unableToDetermine;
     }
 
-    final html.PermissionStatus status = await _permissions!.query(
-      _permissionQuery,
-    );
+    // navigator.permissions.query({ name: "geolocation" })
+    final web.PermissionStatus? status =
+        await _permissions?.query(_permissionQuery.jsify() as JSObject).toDart;
 
-    return status.state != null
-        ? toLocationPermission(status.state)
-        : LocationPermission.denied;
+    return toLocationPermission(status?.state);
```

Comment:
This will result in an exception when the `status` parameter is `null` (this might happen when the user dismisses the permission dialog without specifically granting or denying the permission). 

In these cases the permission request should always return the `denied` permission, informing the user to request permission again.

I think it would be good to use a similar construct as the old statement:
```suggestion
    return status != null
            ? toLocationPermission(status.state)
            : LocationPermission.denied;
```